src/github3/repos/branch.py

```diff
@@ -74,62 +74,64 @@ def protection(self):
         return BranchProtection(json, self)
 
     @decorators.requires_auth
-    def protect(self, enforcement=None, status_checks=None):
-        """Enable force push protection and configure status check enforcement.
+    def protect(self, enforce_admins, required_pull_request_reviews,
+                required_status_checks, restrictions):
```

Comment:
Can't say I'm a fan of this design. But I don't know what else to do: GitHub's API requires us to send a deep object with lots of required keys (see: https://developer.github.com/v3/repos/branches/#update-branch-protection).

Reply:
It's frustrating that they require all of this to start working with branch protections. Instead, I would love for a user to be able to do:

```
protection = branch.protection()
protection.required_status_checks.add_contexts(['integration/travis-ci'])
```

or whatever.

Perhaps what we actually want here is to enable users to do:

```
branch = repo.branch(...)
protection = branch.protection_updater()
protection.enforce_admins().require_pull_request_reviews(...).update()
```

Reply:
Or maybe we want `protect()` to return that? It would be backwards incompatible though.

Reply:
This whole part of the API is atrocious to work with.

Reply:
> It would be backwards incompatible though.

Correct me if I'm wrong, but I wasn't much concerned by this: users' code doesn't currently work, so they can expect to have to change their code in order for it to work. Or should I try to preserve backwards compatibility here?

Reply:
There's always been some amount of attempts towards backwards compatibility. This API grouping though is experimental which makes me lean towards just breaking it and apologizing because we had to.